### DISCUSSION TITLE
How do you kill/cancel a uv_thread_t?

### DISCUSSION THREAD

**Discussion by Qix-** [libuv/libuv#3576] [Q&A]
In the case my server is not capable of reaching a (potentially deadlocked/stale) thread in a short amount of time after signaling for a shutdown, I'd like to forcefully kill/cancel it via something like `pthread_cancel()`. Is this possible with the libuv API? I can't seem to find a similar call in the docs.

**Comment by bnoordhuis** [comment]
There's no libuv API for that. I'd also argue that force-killing a thread is almost always a bad idea.

**Reply by Qix-** [reply]
If I have 256 worker threads trying to shut down gracefully and one of them becomes unreachable/locked for some reason (some transient error) I have three options to continue:

- terminate, putting the other 255 threads' resources into failure modes
- kill the faulty thread (which libuv doesn't allow me to do)
- hang indefinitely, causing all sorts of problems in production

I appreciate the warning label but sometimes these APIs are used legitimately. I don't like when philosophy gets in the way of practicality, personally - is there a reason other than "it shouldn't be used in the general case" that it's not included? Would you accept a PR to add it?

**Reply by bnoordhuis** [reply]
No, because there is no way to get consistent behavior across platforms.
